assets/src/edit-story/app/media/utils/useProcessVideo.js

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * External dependencies
+ */
+import { useCallback } from 'react';
+/**
+ * Internal dependencies
+ */
+import { useStory } from '../../';
+import fetchRemoteFile from './fetchRemoteFile';
+
+function useProcessVideo({ uploadMedia, uploadVideoPoster, updateMedia }) {
+  const { updateElementsByResourceId } = useStory((state) => ({
+    updateElementsByResourceId: state.actions.updateElementsByResourceId,
+  }));
+
+  const updateElement = useCallback(
```

Comment:
This needs a better, less generic name. Perhaps `copyResourceData` or similar?

Reply:
Fixed in https://github.com/google/web-stories-wp/pull/7130/commits/3cd9d965b0b61bbf0c021092776f2a5549420d61